src/protocol/codecs/payload.rs

```diff
@@ -13,15 +13,23 @@ use crate::protocol::{
     invalid_data,
 };
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[allow(clippy::enum_variant_names)]
+#[allow(dead_code)]
 pub enum Payload {
     MsgOfInterest(MsgOfInterest),
     ProposalPayload(Box<ProposalPayload>),
     AgreementVote(Box<AgreementVote>),
+    Ping(PingData),
+    PingReply(PingData),
     NotImplemented,
 }
 
+#[derive(Debug, Clone)]
+pub struct PingData {
+    pub rand_bytes: [u8; 8],
```

Comment:
How about changing name to something more general, like `payload` or sth?

Reply:
Maybe `nonce`? I can leave a comment about how it usually contains random bytes